File: src/test/java/ExpressionsTest.java
import org.junit.jupiter.api.*;

import java.util.Arrays;
import java.util.HashSet;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
@DisplayName("02-Expressions")
class ExpressionsTest {

    @Test
    @Order(1)
    @DisplayName("Create an expression that returns an integer 5")
    void giveMe5() {
        Expressions expressions = new Expressions();
        assertEquals(expressions.giveMe5(), 5);
    }

    @Test
    @Order(2)
    @DisplayName("Create an expression that returns a text 5")
    void giveMeFive() {
        Expressions expressions = new Expressions();

        HashSet<String> valids = new HashSet<>(Arrays.asList(
                "Five",
                "five",
                "5"
        ));

        assertTrue(valids.contains(expressions.giveMeFive()));
    }

    @Test
    @Order(3)
    @DisplayName("Create an expression that returns true")
    void tellTheTrue() {
        Expressions expressions = new Expressions();
        assertTrue(expressions.tellTheTrue());
    }
}

File: src/test/kotlin/ExpressionsTest.kt
import io.mockk.confirmVerified
import io.mockk.spyk
import io.mockk.verify
import kotlin.test.*

internal class ExpressionsTest {
    private val expressions = Expressions()

    @Test
    fun `Create an expression that returns an integer 5`() {
        assertEquals(expressions.giveMe5(), 5)
    }

    @Test
    fun `Create an expression that returns a text 5`() {
        val valids = HashSet(
            mutableListOf(
                "Five",
                "five",
                "5"
            )
        )

        assertTrue(expressions.giveMeFive() in valids)
    }

    @Test
    fun `Create an expression that returns true`() {
        assertTrue(expressions.tellTheTrue())
    }

    @Test
    fun `Just call giveMe5 function inside your function`() {
        val expressions = spyk(Expressions())
        assertEquals(expressions.callGiveMe5(), 5)
        verify {
            expressions.callGiveMe5()
            expressions.giveMe5()
        }

        confirmVerified(expressions)
    }
}